event.go

```diff
@@ -39,13 +39,17 @@ type Event struct {
 
 	// async is an internal flag set if the transition should be asynchronous
 	async bool
+
+	// cancelFunc is called in case the event is canceled.
+	cancelFunc func()
```

Comment:
There's no test covering this field and its execution below – I can move that to the next PR.